### PR TITLE
Use a global DispatchQueue to synchronize parseable outputs and driver error messages

### PR DESCRIPTION
**Pull request by nkcsgexi**
Both of these outputs will be printed to stderr. Without a queue, interleaving them can
lead to malformed parsable output.

rdar://75370288

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]
Makes sense to me. Thanks!

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -307,28 +307,33 @@ public struct Driver {
   /// A collection of all the flags the selected toolchain's `swift-frontend` supports
   @_spi(Testing) public let supportedFrontendFlags: Set<String>
 
+  /// A global queue for emitting non-interrupted messages into stderr
+  public static let stdErrQueue = DispatchQueue(label: "org.swift.driver.emit-to-stderr")
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Driver/Driver.swift:311]: Should this be public? Or just internal?

**Pull request merged**